lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1955,36 +1956,43 @@ private function getValues($value)
             return [$newValue];
         }
 
-        if (is_object($value) && $this->em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
-            $class = $this->em->getClassMetadata(get_class($value));
-            if ($class->isIdentifierComposite) {
-                $newValue = [];
-
-                foreach ($class->getIdentifierValues($value) as $innerValue) {
-                    $newValue = array_merge($newValue, $this->getValues($innerValue));
-                }
-
-                return $newValue;
-            }
-        }
-
-        return [$this->getIndividualValue($value)];
+        return $this->getIndividualValue($value);
     }
 
     /**
      * Retrieves an individual parameter value.
      *
      * @param mixed $value
      *
-     * @return mixed
+     * @return       array<mixed>
+     *
+     * @psalm-return list<mixed>
      */
     private function getIndividualValue($value)
```

Comment:
I think method name is no longer correct. It was used only as a fallback when it was clear there is only one value

Reply:
I am not sure, it still returns the parameters for an individual value (which can be an entity with a composite id), whereas the method "above" also handles the case of arrays of individual values.